test/distributed/fsdp/test_fsdp_sharded_grad_scaler.py

```diff
@@ -196,6 +203,168 @@ def test_fsdp_ddp_parity_with_grad_scaler(
+            MLP(lin_dim, device),
+            MLP(lin_dim, device, dim_multiplier=3),
+        )
+
+        if build_extra_ddp:
+            ref_model = copy.deepcopy(model).cuda()
+            replicate(ref_model, device_ids=[self.rank])
+            ref_optim = torch.optim.SGD(ref_model.parameters(), lr=1e-2)
+
+        fsdp_kwargs = {
+            "use_orig_params": use_orig_params,
+            "cpu_offload": cpu_offload,
+            "auto_wrap_policy": ModuleWrapPolicy({MLP}),
+        }
+
+        model = FSDP(model, **fsdp_kwargs)
+        optim = torch.optim.SGD(model.parameters(), lr=1e-2)
+
+        if build_extra_ddp:
+            return model, optim, ref_model, ref_optim
+        else:
+            return model, optim
+
+    @skip_if_lt_x_gpu(2)
+    def test_sharded_grad_scaler_found_inf(self):
+        self.run_subtests(
+            {
+                "use_orig_params": [False, True],
+                "cpu_offload": [
+                    CPUOffload(offload_params=True),
+                    CPUOffload(offload_params=False),
+                ],
+            },
+            self._test_sharded_grad_scaler_found_inf,
+        )
+
+    def _test_sharded_grad_scaler_found_inf(
+        self,
+        use_orig_params: bool,
+        cpu_offload: CPUOffload,
+    ):
+        lin_dim = 4
+        model, optim, ref_model, ref_optim = self._build_model_and_optim(
+            lin_dim,
+            build_extra_ddp=True,
+            cpu_offload=cpu_offload,
+            use_orig_params=use_orig_params,
+        )
+        grad_scaler = ShardedGradScaler(init_scale=2.0)
+        ref_grad_scaler = torch.cuda.amp.GradScaler(init_scale=2.0)
+        scaled_losses: List[torch.Tensor] = []
+        torch.manual_seed(42 + self.rank + 1)
+
+        for iter in range(10):
+            x = torch.rand((32, lin_dim), device="cuda")
+            for _model, _optim, _grad_scaler in (
+                (ref_model, ref_optim, ref_grad_scaler),
+                (model, optim, grad_scaler),
+            ):
+                _optim.zero_grad()
+                loss = _model(x).sum()
+                scaled_loss = _grad_scaler.scale(loss)
+                scaled_losses.append(scaled_loss)
+                scaled_loss.backward()
+
+                orig_params = [
+                    param.detach().clone()
+                    for param in _model.parameters()
+                    if param.grad is not None
+                ]
+
+                should_find_inf = iter % 2 == 0
+                if should_find_inf and (
+                    _model == ref_model or (_model == model and dist.get_rank() != 0)
+                ):
+                    # other ranks should find infs from rank 0
+                    # after collectives
+                    for param in _model.parameters():
+                        if param.grad is None:
+                            continue
+                        param.grad.data.fill_(float("inf"))
```

Comment:
```suggestion
                        param.grad.data.fill_(float("inf"))
```
nit: Do you need the `.data`? If so, maybe we prefer `.detach()`.

Reply:
no need for .data. thus changed to `param.grad.fill_`. In-place ops can be performed on non-leaf tensors